### PR TITLE
add date, refactor time to clock

### PR DESCRIPTION
**Pull request by stefanmunz**
adds the date command, also extracted time helper into an extra class, because they are used in date and clock.

### CODE REVIEW

File: clock.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package commands
+
+// Now returns the current time as a string
```
Comment (pietbrauer) [inline clock.go]: `Now` describes the command name so this should be renamed as well.

File: clock_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package commands
+
+import "testing"
+
+func TestNow12HoursAM(t *testing.T) {
```
Comment (pietbrauer) [inline clock_test.go]: Should be `TestClockWith12HoursAM`

File: clock_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	ForceTimeTo(1415694000)
+	const outAM = "8:20AM"
+	if y := Clock(false); y != outAM {
+		t.Errorf("Clock() = %v, want %v", y, outAM)
+	}
+}
+
+func TestNow12HoursPM(t *testing.T) {
```
Comment (pietbrauer) [inline clock_test.go]: Should be `TestClock12HoursPM`

File: clock_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	ForceTimeTo(1415737200)
+	const outPM = "8:20PM"
+	if x := Clock(false); x != outPM {
+		t.Errorf("Clock() = %v, want %v", x, outPM)
+	}
+}
+
+func TestNow24HoursAM(t *testing.T) {
```
Comment (pietbrauer) [inline clock_test.go]: Should be `TestClock24HoursAM`

File: clock_test.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	ForceTimeTo(1415694000)
+	const outAM = "08:20"
+	if y := Clock(true); y != outAM {
+		t.Errorf("Clock() = %v, want %v", y, outAM)
+	}
+}
+
+func TestNow24HoursPM(t *testing.T) {
```
Comment (pietbrauer) [inline clock_test.go]: Should be `TestClock24HoursPM`

File: date.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package commands
+
+// date returns the current date as a string
```
Comment (pietbrauer) [inline date.go]: Date should start with a capital D

**Comment by pietbrauer** [comment]
Please also consider the [linting failures](https://drone.io/github.com/jarvisio/commands/10):

```
clock.go:3:1: comment on exported function Clock should be of the form "Clock ..."
date.go:3:1: comment on exported function Date should be of the form "Date ..."
time_helper.go:11:1: exported function ForceTimeTo should have comment or be unexported
```

File: time_helper.go [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"time"
+)
+
+var currentTime = func() time.Time {
+	return time.Now()
+}
+
+func ForceTimeTo(unix int) {
```
Comment (pietbrauer) [inline time_helper.go]: This method should be private according to the linter, so change it to `forceTimeTo`.
`time_helper.go:11:1: exported function ForceTimeTo should have comment or be unexported`

**Comment by stefanmunz** [comment]
How did you lint the code? I cannot register at drone.io at the moment, did not work.

**Comment by pietbrauer** [comment]
Just `make check`

**Comment by stefanmunz** [comment]
ok I make'd the check :)

**Pull request merged**